[crash] AssertionError: @AssumeAssertion(nullness): currentNode is nonull if isRunning

```java
import java.util.List;
import java.util.function.Function;

abstract class T {

  interface V {}

  abstract <F, T> Iterable<T> transform(Iterable<F> i, Function<? super F, ? extends T> f);

  abstract <E> List<E> copyOf(Iterable<? extends E> e);

  List<V> generateAppSuggestions(List<Integer> xs) {
    return copyOf(transform(xs, x -> new V() {}));
  }
}
```

```
./checker-framework-3.9.0/checker/bin/javac -processor Nullness T.java
error: SourceChecker.typeProcess: unexpected Throwable (AssertionError) while processing T.java; message: @AssumeAssertion(nullness): currentNode is nonull if isRunning.
  ; The Checker Framework crashed.  Please report the crash.
  Compilation unit: T.java
  Last visited tree at line 4 column 1:
  abstract class T {
  Exception: java.lang.AssertionError: @AssumeAssertion(nullness): currentNode is nonull if isRunning.; java.lang.AssertionError: @AssumeAssertion(nullness): currentNode is nonull if isRunning.
        at org.checkerframework.dataflow.analysis.AbstractAnalysis.getValue(AbstractAnalysis.java:187)
        at org.checkerframework.dataflow.analysis.AbstractAnalysis.getValue(AbstractAnalysis.java:280)
        at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.getInferredValueFor(GenericAnnotatedTypeFactory.java:1738)
        at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.addComputedTypeAnnotations(GenericAnnotatedTypeFactory.java:1689)
        at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.addComputedTypeAnnotations(GenericAnnotatedTypeFactory.java:1636)
        at org.checkerframework.framework.type.AnnotatedTypeFactory.getAnnotatedTypeFromTypeTree(AnnotatedTypeFactory.java:1118)
        at org.checkerframework.framework.type.SupertypeFinder$SupertypeFindingVisitor.supertypesFromTree(SupertypeFinder.java:271)
        at org.checkerframework.framework.type.SupertypeFinder$SupertypeFindingVisitor.visitDeclared(SupertypeFinder.java:189)
        at org.checkerframework.framework.type.SupertypeFinder.directSuperTypes(SupertypeFinder.java:52)
        at org.checkerframework.framework.type.AnnotatedTypeMirror$AnnotatedDeclaredType.directSuperTypes(AnnotatedTypeMirror.java:982)
        at org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Declared(AsSuperVisitor.java:331)
        at org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Declared(AsSuperVisitor.java:29)
        at org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:314)
        at org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:65)
        at org.checkerframework.framework.type.AsSuperVisitor.visit(AsSuperVisitor.java:98)
        at org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Typevar(AsSuperVisitor.java:379)
        at org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Typevar(AsSuperVisitor.java:29)
        at org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:344)
        at org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:65)
        at org.checkerframework.framework.type.AsSuperVisitor.visit(AsSuperVisitor.java:98)
        at org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Wildcard(AsSuperVisitor.java:398)
        at org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Wildcard(AsSuperVisitor.java:29)
        at org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:352)
        at org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:65)
```

--- a/dataflow/src/main/java/org/checkerframework/dataflow/analysis/AbstractAnalysis.java
+++ b/dataflow/src/main/java/org/checkerframework/dataflow/analysis/AbstractAnalysis.java
@@ -76,6 +76,8 @@ public abstract class AbstractAnalysis<
      *   !isRunning &rArr; (currentNode == null)
      * </pre>
      */
+    // currentNode == null when isRunning is true.
+    // See https://github.com/typetools/checker-framework/issues/4115
     protected @InternedDistinct @Nullable Node currentNode;
 
     /**

--- a/dataflow/src/main/java/org/checkerframework/dataflow/analysis/AbstractAnalysis.java
+++ b/dataflow/src/main/java/org/checkerframework/dataflow/analysis/AbstractAnalysis.java
@@ -184,10 +184,10 @@ public abstract class AbstractAnalysis<
     @Override
     public @Nullable V getValue(Node n) {
         if (isRunning) {
-            assert currentNode != null
-                    : "@AssumeAssertion(nullness): currentNode is nonull if isRunning.";
             // we don't have a org.checkerframework.dataflow fact about the current node yet
-            if (currentNode == n || (currentTree != null && currentTree == n.getTree())) {
+            if (currentNode == null
+                    || currentNode == n
+                    || (currentTree != null && currentTree == n.getTree())) {
                 return null;
             }
             // check that 'n' is a subnode of 'node'. Check immediate operands

--- a/framework/tests/all-systems/Issue4115.java
+++ b/framework/tests/all-systems/Issue4115.java
@@ -1,0 +1,15 @@
+import java.util.List;
+import java.util.function.Function;
+
+abstract class Issue4115 {
+
+    interface V {}
+
+    abstract <F, T> Iterable<T> transform(Iterable<F> i, Function<? super F, ? extends T> f);
+
+    abstract <E> List<E> copyOf(Iterable<? extends E> e);
+
+    List<V> generateAppSuggestions(List<Integer> xs) {
+        return copyOf(transform(xs, x -> new V() {}));
+    }
+}

